fe-code-review/code_for_review.js

```diff
@@ -0,0 +1,142 @@
+                        };
+                        logDebug('Send config:', sendConfig);
+
+                        try {
+                            await db.setEmailLog({collectorEmail, idCollector, idUser, requestKey})
+                        } catch (e) {
+                            logDebug('extract() setEmailLog error=', e);
+                        }
+
+                        /* send email */
+                        const resp = await email.send(sendConfig, config.projects[project].email.apiKey);
+                        logDebug('extract() resp=', resp);
+
+                        // update DB with result
+                        await db.setUserCollectorRequestKeyRes(requestKey, idUser, idCollector, resp);
+
+                        if (!sentStatus[collectorName])
+                            sentStatus[collectorName] = {};
+                        sentStatus[collectorName][collectorEmail] = resp;
+
+                        if (!resp) {
+                            logError('extract() Sending email failed: ', resp);
+                        }
+                    }
+                }
+                await db.updateStatus(requestID, 100, '');
+
+                logDebug('FINAL SENT STATUS:');
+                console.dir(sentStatus, {depth: null});
+
+                //if (!allSent)
+                //return res.status(500).json({requestID, message: 'Failed sending email'});
+
+                await db.updateStatus(requestID, 500, '');
+
+                /* prepare summary email */
+                const summaryConfig = {
+                    //bcc: [{ email: 'tomas@inkassoregisteret.com', name: 'Tomas' }],
+                    sender: config.projects[project].email.sender,
+                    replyTo: config.projects[project].email.replyTo,
+                    subject: 'Oppsummering Kravsforespørsel',
+                    templateId: config.projects[project].email.template.summary,
+                    params: {
+                        collectors: sentStatus,
+                    },
+                    tags: ['summary'],
+                    to: [{ email: 'tomas@upscore.no' , name: 'Tomas' }], // FIXXX: config.projects[project].email.sender
+                };
+                logDebug('Summary config:', summaryConfig);
+
+                /* send email */
+                //const respSummary = await email.send(sendConfig, config.projects[project].email.apiKey);
+                //logDebug('extract() summary resp=', respSummary);
+
+                await db.updateStatus(requestID, 900, '');
+            }
+            await db.updateStatus(requestID, 999, '');
+            return res.json({requestID, step: 999, status: 'DONE', message: 'Done sending emails...'});
+        } else
+            return res.status(500).json({requestID, message: 'Missing requried input (requestID, project, file)'});
+    }
+    res.status(500).json({requestID: '', message: 'Missing requried input (form data)'});
```

Comment:
This code should be a little bit optimized and divided into smaller functions. Also it makes sense to check conditions at the very beginning to prevent some excessive logic.

Not enough comments to briefly understand the logic